zrml/prediction-markets/src/tests.rs

```diff
@@ -1503,37 +1515,30 @@ fn full_scalar_market_lifecycle() {
         let ztg_bal_eve = Balances::free_balance(&EVE);
         assert_eq!(ztg_bal_charlie, 98750 * CENT); // 75 (LONG) + 12.5 (SHORT) + 900 (balance)
         assert_eq!(ztg_bal_eve, 1000 * BASE);
-        assert!(event_exists(Event::TokensRedeemed(
+        System::assert_has_event(Event::PredictionMarkets(crate::Event::TokensRedeemed(
             0,
             Asset::ScalarOutcome(0, ScalarPosition::Long),
             100 * BASE,
             75 * BASE,
-            CHARLIE
+            CHARLIE,
         )));
-        assert!(event_exists(Event::TokensRedeemed(
+        System::assert_has_event(Event::PredictionMarkets(crate::Event::TokensRedeemed(
             0,
             Asset::ScalarOutcome(0, ScalarPosition::Short),
             50 * BASE,
             1250 * CENT, // 12.5
-            CHARLIE
+            CHARLIE,
         )));
 
         assert_ok!(PredictionMarkets::redeem_shares(Origin::signed(EVE), 0));
         let ztg_bal_eve_after = Balances::free_balance(&EVE);
         assert_eq!(ztg_bal_eve_after, 101250 * CENT); // 12.5 (SHORT) + 1000 (balance)
-        assert!(!event_exists(Event::TokensRedeemed(
```

Comment:
Event should not exist here. You wrote this @maltekliemann one month ago. Is there a specific reason. Maybe use `!assert_has_event`?

Reply:
Testing for an event _not_ to exist seems like a bad idea in hindsight. Every dispatch should not emit most events.

But... If there was a buildin function that could test for the lack of events, I would have kept the test. But `assert_has_event` doesn't return a bool, it just panicks if the event doesn't exist, so `!assert_has_event` won't compile.